docker/Dockerfile

```diff
@@ -20,18 +24,13 @@ COPY cheribuild.json /root/.config/cheribuild.json
 
 # deps to build QEMU+elftoolchain:
 RUN apt-get update && apt-get install -y \
-  libtool pkg-config python-minimal autotools-dev automake autoconf libglib2.0-dev libpixman-1-dev \
+  libtool pkg-config python2-minimal autotools-dev automake autoconf libglib2.0-dev libpixman-1-dev \
   bison groff-base libarchive-dev flex
 
-# INSTALL clang 8.0
-RUN wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
-RUN echo "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-8 main" > /etc/apt/sources.list.d/llvm.list
-# RUN echo "deb http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu trusty main" > /etc/apt/sources.list.d/r-toolchain.list
-# RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 1E9377A2BA9EF27F
-RUN apt-get update && apt-get install -y clang-8 lld-8
-
 RUN apt-get update && apt-get install -y cmake
 
+RUN addgroup --gid 12345 cheri && adduser --uid 12345 --ingroup cheri cheri
```

Comment:
And if someone has UID/GID 12345 how do they use Dockerfile.useradd without conflicts?

Reply:
I recognize that this might be adding more moving parts, and so the opposite of what we want to do, but... https://github.com/containers/fuse-overlayfs has support for remapping [UG]IDs seen on the backing store(s) to different [UG]IDs as seen by the mount.  Since the in-guest vnops use the synthetic UIDs reported by the overlayfs, there's no need to synchronize UIDs inside and outside the container, and it can be made to work either with or without UID namespaces.  I think this means we could establish a mount with in-container 12345 transparently mapped to the out-of-container user on any volumes that get mounted?

Perhaps worth a look, perhaps excessive, or perhaps missing the point entirely?